### DISCUSSION TITLE
Using pug doesn't set index.html as root path

### DISCUSSION THREAD

**Discussion by vncastanheira** [parcel-bundler/parcel#6012] [Q&A] [answered]
Basically if I run `parcel index.html` and go to the root path (e.g. `localhost:123/`) , it will load the `index.html` page. 

But if I use pug and do something like `parcel index.pug`, I have to go to `localhost:123/index.html` in order to open the page.

Why does this happen?

**Comment by andrewspear** [comment]
I have a similar issue. Running `parcel src/html/*.html` and then browsing to `localhost:1234` gives a 404 but browsing to `localhost:1234/index.html` works as expected.

There is a note [on this page](https://v2.parceljs.org/features/cli/) which says:

> If you have specified multiple HTML entry points and none of them is has the output path /index.html, the dev server will respond to localhost:1234/ with a 404, since Parcel doesn't know which HTML bundle is the index. In this case, load the file directly e.g. localhost:1234/a.html and localhost:1234/b.html

That sounds like the issue, but I'd like to know how to work around it. I'm working with a third party app which needs to access the root domain but currently it returns the 404.

Is there a way to customise the built-in server to define the default file? In `package.json` we specify a `main` attribute which is supposed to be the entry point for the application, perhaps Parcel could recognise that and load it at the root path 🤔 Any suggestions would be appreciated.

**Reply by DeMoorJasper** [reply]
You can use this to change the server logic: https://v2.parceljs.org/features/api-proxy/

**Comment by andrewspear** [comment, marked as answer]
@vncastanheira I have just upgraded the project to Parcel v2 and it solved the issue for me.
